### PR TITLE
Added script bin/serve_more

### PR DESCRIPTION
**Pull request by stappersg**
Script for making it easy that files as
```text
  /usr/lib/debian-installer/images/11/arm64/text/debian-installer/arm64/initrd.gz
  /usr/lib/debian-installer/images/11/arm64/text/debian-installer/arm64/linux
```
on the shoelaces server can be served as
```text
  http://shoelaces.lan:8081/static/di/arm64/initrd
  http://shoelaces.lan:8081/static/di/arm64/linux
```

Signed-off-by: Geert Stappers <stappers@stappers.it>

### CODE REVIEW

**Comment by stappersg** [comment]
FYI: _sales pitch_ is planned for later this week.

**Comment by stappersg** [comment]
(The _sales pitch_ is pending / dangling.)

When the script is shipped as `web/bin/serve_more` we will need
```text
--- a/test/integ-test/expected-results/static.html
+++ b/test/integ-test/expected-results/static.html
@@ -1,4 +1,5 @@
 <pre>
+<a href="bin/">bin/</a>
 <a href="css/">css/</a>
 <a href="fonts/">fonts/</a>
 <a href="img/">img/</a>
```

**Comment by stappersg** [comment]
The benefit of `serve_more` script is avoiding external website by serving more with `shoelaces`.

That as _a diff_:
```diff
--- debian.ipxe.slc	2021-01-13 20:37:40.000000000 +0100
+++ Debian.ipxe.slc	2023-01-22 16:37:23.898797403 +0100
@@ -1,16 +1,19 @@
-{{define "debian.ipxe" -}}
+{{define "Debian.ipxe" -}}
 #!ipxe
 
 echo This automatically overwrites data!
 echo Debian {{.release}}
 
-set mirror http://ftp.debian.org/debian/dists/{{.release}}/main/installer-amd64/current/images/netboot/debian-installer/amd64
+set mirror http://{{.baseURL}}/static/netboot/debian/12/amd64
 
 chain http://{{.baseURL}}/configs/linux.cfg?hostname={{.hostname}}
 
 imgfree
```
So `mirror` is _shoelaces_,  **not** an outside website.

Example of a file that `serve_more` can process:
```text
#
# serve files from packages 
#   debian-installer-11-netboot-amd64
#   debian-installer-11-netboot-arm64
# under a nicer URL
#
#
di/arm64/initrd.gz  /usr/lib/debian-installer/images/11/arm64/text/debian-installer/arm64/initrd.gz
di/arm64/linux   /usr/lib/debian-installer/images/11/arm64/text/debian-installer/arm64/linux

di/amd64/initrd.gz  /usr/lib/debian-installer/images/11/amd64/text/debian-installer/amd64/initrd.gz
di/amd64/linux   /usr/lib/debian-installer/images/11/amd64/text/debian-installer/amd64/linux
#
# l l
```

**Comment by stappersg** [comment]
Please review this merge request.

File: web/bin/serve_more [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+fi
+
+TOP=${PWD}  # Top of web directory, the 'static-dir=' value
+
+while read TOSERVE CONTENT
+do
+	echo ${TOSERVE} ${CONTENT}
+	TSDIR=$( dirname ${TOSERVE} )
+	TSFN=$( basename ${TOSERVE} )
+	cd ${TOP}
```
Comment (raul-te) [inline web/bin/serve_more:26]: In scripts, it's preferable to use `pushd` and `popd`, a feature known as the [Directory Stack](https://www.gnu.org/software/bash/manual/html_node/The-Directory-Stack.html). That way, you won't surprise a user by inadvertently changing directories.

Anyway, I think that in this situation you can skip changing directories and use absolute paths in `ln`.

File: web/bin/serve_more [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (raul-te) [inline web/bin/serve_more:1]: It seems this is a script meant to be run by the Shoelaces administrator when setting up the files to serve. Hence, I don't think we should be hosting this file in the Shoelaces webserver. A new `utils` directory in the root of the repository seems more appropiate.

File: web/bin/serve_more [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#
+#  serve more
+#  Script for making it easy that shoelaces
+#  can serve more static content.
+#  That content came to various way to the shoelaces server
+#  ( downloads, Linux distro packages, self build )
+#  This script creates symbolic links to it.
```
Comment (raul-te) [inline web/bin/serve_more:8]: The description is confusing. Can you rephrase it, please?

**Comment by stappersg** [comment]
For what it is worth:   I think this merge request can be closed.  I left it open for comparing with #34

**Comment by raul-te** [comment]
Closing in favor of #34
